docs/API_DESIGN.md

```diff
@@ -90,11 +92,29 @@ CLASS VerifiableCredential
   PUBLIC DATA credentialSubject: CredentialSubject
 
   CONSTRUCTOR create(issuer: Issuer, credential_subject: CredentialSubject, options: CreateOptions)
+  CONSTRUCTOR decode(vc_jwt: string, options: VerifiableCredentialDecodeOptions)
 
-  CONSTRUCTOR(vcjwt: string)
-  CONSTRUCTOR(vcjwt: string, verifier: Verifier)
-  METHOD sign(bearer_did: BearerDid): string
-  METHOD sign_with_signer(key_id: string, signer: Signer): string
+  METHOD sign(options: VerifiableCredentialSignOptions)
+```
+
+##### `VerifiableCredentialDecodeOptions`
+
+```pseudocode!
+CLASS VerifiableCredentialDecodeOptions
+  PUBLIC DATA verify: bool?
+```
+
+##### `VerifiableCredentialSignOptions`
+
+> [!NOTE]
+>
+> If a `signer` is specified, then it takes first priority. Else, the `bearer_did` will be used. If a `verification_method_id` is not specified then the first Verification Method in the `bearer_did` will be utilized. Given the `signer` IS NOT specified and the `bearer_did` IS specified, then the `issuer` ID defined in the call to `create()` MUST be equal to value of the `bearer_did`'s DID URI.
```

Comment:
this is a little hard to follow. might be better to have multiple sign methods that take different arguments

Reply:
It looks like the two ways to sign are
1. `sign(signer: Signer)`
2. `sign_with_did(bearer_did: BearerDid, verification_method_id: Option<String>)`

Reply:
I'm cool with breaking it up, however with the caveat of this instead

`sign_with_did(bearer_did: BearerDid, options: VerifiableCredentialSignWithDidOptions)`

The reason being, across all of our SDKs, there are a variety of opinions as to how we should _query_ for the given signer, so we can't commit to only supporting `verification_method_id`.